src/core/terms.ml

```diff
@@ -242,11 +244,12 @@ type term =
   ; meta_value : (term, term) Bindlib.mbinder option ref
   (** Definition of the metavariable, if known. *) }
 
-(** Comparison function on metavariables. *)
-let cmp_meta : meta -> meta -> int = fun m1 m2 -> m1.meta_key - m2.meta_key
+module Meta = struct
+  type t = meta
+  let compare m1 m2 = m1.meta_key - m2.meta_key
+end
 
-(** Equality on metavariables. *)
-let eq_meta : meta -> meta -> bool = fun m1 m2 -> m1.meta_key = m2.meta_key
```

Comment:
I would still keep this definition. The `compare` function that you expose in the `Meta` module does not make much sense outside of building sets (there is no useful order on metavariables). I would hence suggest defining the following to avoid exporting your `Meta` module:
```ocaml
module MetaSet = Set.Make(
  struct
    type t = meta
    let compare m1 m2 = m1.meta_key - m2.meta_key
  end)
```

Reply:
But eq_meta is used nowhere at the moment...

Reply:
It does not hurt to have it, but whatever. However the second point is important.

Reply:
Meta.compare is used in proof.ml

Reply:
That was not the point, I still think such a function should not be exposed. And it should also not be exposed by `Proofs.Goal`. What I would propose is to expose a set interface as part of the `Proofs.Goal` module as well. I will make suggestion in the relevant part of the code.

Reply:
My point still stands here.